Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -1136,8 +1137,18 @@ namespace ScriptCanvasEditor
         AZ_Assert(slot, "A valid slot must be provided");
         if (slot)
         {
+            
             m_slotTypeSelector = new SlotTypeSelectorWidget(GetActiveScriptCanvasId(), this); // Recreate the widget every time because of https://bugreports.qt.io/browse/QTBUG-76509
-            m_slotTypeSelector->PopulateVariablePalette(m_variablePaletteTypes);
+            if (azrtti_istypeof<const ScriptCanvas::Nodes::Math::MathExpression*>(slot->GetNode()))
+            {
+                AZStd::unordered_set<AZ::Uuid> variableTypes = { ToAZType(ScriptCanvas::Data::Type::Number()), ToAZType(ScriptCanvas::Data::Type::Vector3()) };
+
+                m_slotTypeSelector->PopulateVariablePalette({}, &variableTypes);
+            }
+            else
+            {
+                m_slotTypeSelector->PopulateVariablePalette(m_variablePaletteTypes);
+            }
```

Comment:
If you pass the node through the SlotTypeSelectorWidget constructor, you can remove the special case handling of MathExpression in MainWindow.cpp. While I don't love either approach, I would rather keep specific nodes (especially one that is dependent on a separate gem) out of MainWindow.cpp

A more futureproof approach could be to make a simple factory bus to create the SlotTypeSelector, this way you can do something like:
`NodeRequestBus::EventResult(m_slotTypeSelector, slot->GetNode(), &NodeRequests::CreateSlotTypeSelector, m_variablePaletteTypes);`

Then in the default implementation of NodeRequests::CreateSlotTypeSelector you construct it as you do in the else case, and for the MathExpression node, you go and connect that bus and let it return the trimmed down slot type selector.

Reach out if you need clarification.

Reply:
hi, I was wondering if there was any reference I could use to achieve this. I never had to use bus before, so I wasn't sure where to start off